### PR TITLE
Add thread group swizzling feature for compute shaders.

### PR DESCRIPTION
**Pull request by ruanjm**
This feature generates new thread group id based on system thread group id for user. The algorithm is generating morton code from the flat system thread group id. So that the order of new id would look like z-order Lebesgue curve.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JaxLinAMD** [comment]
ok to test( <-- token to trigger jenkins)

**Comment by JaxLinAMD** [comment]
add to whitelist (<- token to add ruanjm to whitelist)

File: include/vkgcDefs.h [hunk 1]
```diff
@@ -328,6 +328,15 @@ struct ExtendedRobustness {
                            ///  valid to access and acts as if the descriptor is bounded to nothing.
 };
 
+/// Represents the tiling modes for compute shader thread group swizzling
```
Comment (amdrexu) [inline include/vkgcDefs.h:331]: Please bump the version to 45.5 and add a comment in the beginning of this file (mention that you add a new enum type for CS swizzling).

LLPC_INTERFACE_MINOR_VERSION 4 -> LLPC_INTERFACE_MINOR_VERSION 5

**Review by amdrexu — APPROVED** [review]
LGTM.
Next time, when you make any code update. You can use `git commit --amend -C head` to modify your changes and submit them, rather than using `git commit`. This will not generate new commit for this pull request.

**Pull request closed without merging**